No Longer Able to Read State File

Something happened within the last 24 hours to where my Azure DevOps terraform plan task can no longer load or read the statefile.  Apparently there is an extra command line argument now.

Was there a commit recently that could be causing this?

![image](https://user-images.githubusercontent.com/20479122/143071911-b9de459b-67c5-48fe-9f60-4711bfbb8373.png)

Update: Same thing happens with Terraform: Apply.   Terraform Init appears to work okay, but with Plan and Apply I end up with the following error:

`##[error]Error: Failed rmRF: EPERM: operation not permitted, unlink 'D:\a\r1\a\temp-output-9ff0c9db-acac-4a32-b1fc-fcf564b74476.json'`

Also seeing this error, same task ran successfully 12 hours ago. 

`task: TerraformTaskV2@2
      displayName: Terraform Plan
      inputs:
        provider: 'azurerm'
        command: 'plan'
        workingDirectory: $(Pipeline.Workspace)/terraform
        commandOptions: -out=plan -var-file=$(Pipeline.Workspace)/terraform/envvalues/GA/stag.tfvars
        environmentServiceNameAzureRM: redacted`

`Plan: 5 to add, 0 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────

Saved the plan to: plan

To perform exactly these actions, run the following command to apply:
    terraform apply "plan"
C:\a\_tool\terraform\1.0.9\x64\terraform.exe version
Terraform v1.0.9
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.80.0
+ provider registry.terraform.io/hashicorp/helm v2.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1

Your version of Terraform is out of date! The latest version
is 1.0.11. You can update by downloading from https://www.terraform.io/downloads.html
##[error]Error: Failed rmRF: EPERM: operation not permitted, unlink 'C:\a\1\s\temp-plan-0c5d078f-ab26-4c11-95e1-e17ed0fb59cc.txt'`

Same issue here task ran successfully before the Node 10 update. 

2021-11-23T19:50:45.5240377Z ------------------------------------------------------------------------
2021-11-23T19:50:45.5240703Z 
2021-11-23T19:50:45.5241130Z This plan was saved to: marketplace.tfplan
2021-11-23T19:50:45.5241484Z 
2021-11-23T19:50:45.5242457Z To perform exactly these actions, run the following command to apply:
2021-11-23T19:50:45.5243060Z     terraform apply "marketplace.tfplan"
2021-11-23T19:50:45.5243321Z 
2021-11-23T19:50:45.5586656Z [command]C:\hostedtoolcache\windows\terraform\0.12.9\x64\terraform.exe version
2021-11-23T19:50:46.2678579Z Terraform v0.12.9
2021-11-23T19:50:46.2679485Z + provider.azurerm v2.56.0
2021-11-23T19:50:46.2679965Z 
2021-11-23T19:50:46.2680615Z Your version of Terraform is out of date! The latest version
2021-11-23T19:50:46.2681392Z is 1.0.11. You can update by downloading from www.terraform.io/downloads.html
2021-11-23T19:51:03.4267542Z **##[error]Error: Failed rmRF: EPERM: operation not permitted, unlink 'D:\a\r1\a\temp-plan-1022a0c7-c7f7-4b54-bf16-0670d5c2447f.txt'**

--- a/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV1/package.json
+++ b/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV1/package.json
@@ -13,7 +13,8 @@
     "@types/node": "^10.17.0",
     "@types/q": "^1.5.0",
     "@types/uuid": "^8.3.1",
-    "azure-pipelines-task-lib": "^3.1.7"
+    "azure-pipelines-task-lib": "^3.1.7",
+    "del": "^6.0.0"
   },
   "devDependencies": {
     "typescript": "^4.0.0"

--- a/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV1/src/base-terraform-command-handler.ts
+++ b/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV1/src/base-terraform-command-handler.ts
@@ -5,6 +5,7 @@ import tasks = require('azure-pipelines-task-lib/task');
 import path = require('path');
 import * as uuidV4 from 'uuid/v4';
 const fs = require('fs');
+const del = require('del');
 
 export abstract class BaseTerraformCommandHandler {
     providerName: string;
@@ -178,15 +179,22 @@ export abstract class BaseTerraformCommandHandler {
 
             // Delete all the files that are not needed any further
             if (tasks.exist(binaryPlanFilePath)) {
-                tasks.rmRF(binaryPlanFilePath);
+                
+                (async () => {
+                    await del([binaryPlanFilePath]);
+                })();
             }
 
             if (tasks.exist(tempFileForPlanOutput)) {
-                tasks.rmRF(tempFileForPlanOutput);
+                (async () => {
+                    await del([tempFileForPlanOutput]);
+                })();
             }
 
             if (tasks.exist(tempFileForJsonPlanOutput)) {
-                tasks.rmRF(tempFileForJsonPlanOutput);
+                (async () => {
+                    await del([tempFileForJsonPlanOutput]);
+                })();
             }
 
         } else {
@@ -257,7 +265,9 @@ export abstract class BaseTerraformCommandHandler {
 
         // Delete the temp file as it is not needed further
         if (tasks.exist(tempFileForJsonOutputVariables)) {
-            tasks.rmRF(tempFileForJsonOutputVariables);
+            (async () => {
+                await del([tempFileForJsonOutputVariables]);
+            })();
         }
     }
 

--- a/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/package.json
+++ b/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/package.json
@@ -9,11 +9,12 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@types/node": "^10.17.0",
     "@types/mocha": "2.2.48",
+    "@types/node": "^10.17.0",
     "@types/q": "^1.5.0",
     "@types/uuid": "^8.3.1",
-    "azure-pipelines-task-lib": "^3.1.7"
+    "azure-pipelines-task-lib": "^3.1.7",
+    "del": "^6.0.0"
   },
   "devDependencies": {
     "typescript": "^4.0.0"

--- a/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/src/base-terraform-command-handler.ts
+++ b/Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/src/base-terraform-command-handler.ts
@@ -5,6 +5,7 @@ import tasks = require('azure-pipelines-task-lib/task');
 import path = require('path');
 import * as uuidV4 from 'uuid/v4';
 const fs = require('fs');
+const del = require('del');
 
 export abstract class BaseTerraformCommandHandler {
     providerName: string;
@@ -179,15 +180,21 @@ export abstract class BaseTerraformCommandHandler {
 
             // Delete all the files that are not needed any further
             if (tasks.exist(binaryPlanFilePath)) {
-                tasks.rmRF(binaryPlanFilePath);
+                (async () => {
+                    await del([binaryPlanFilePath]);
+                })();
             }
 
             if (tasks.exist(tempFileForPlanOutput)) {
-                tasks.rmRF(tempFileForPlanOutput);
+                (async () => {
+                    await del([tempFileForPlanOutput]);
+                })();
             }
 
             if (tasks.exist(tempFileForJsonPlanOutput)) {
-                tasks.rmRF(tempFileForJsonPlanOutput);
+                (async () => {
+                    await del([tempFileForJsonPlanOutput]);
+                })();
             }
 
         } else {
@@ -258,7 +265,9 @@ export abstract class BaseTerraformCommandHandler {
 
         // Delete the temp file as it is not needed further
         if (tasks.exist(tempFileForJsonOutputVariables)) {
-            tasks.rmRF(tempFileForJsonOutputVariables);
+            (async () => {
+                await del([tempFileForJsonOutputVariables]);
+            })();
         }
     }
 

